### PR TITLE
chore(selector): refactored the selector transformer

### PR DESCRIPTION
**Pull request by anikethsaha**
- added `postcss-selector-parser`
- Transforming HTML tags, attributes, and selected pseudo classes

### CODE REVIEW

File: src/selector.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const caseSensitivePseudoClassesSet = new Set(caseSensitivePseudoClassesLists);
+
+function getParsed(selectors, callback) {
+  return parser(callback).processSync(selectors);
+}
+
+export const transformer = (str) =>
+  getParsed(str, (sel) => {
+    sel.walkTags((selNode) => {
+      selNode.value = selNode.value.toLowerCase();
```
Comment (evilebottnawi) [inline src/selector.js]: Let's use https://github.com/sindresorhus/html-tags and lowercase only known

File: src/selector.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      return selNode;
+    });
+    sel.walkAttributes((selNode) => {
+      selNode._attribute = selNode._attribute.toLowerCase();
+    });
+    sel.walkPseudos((selNode) => {
+      const normalizeVal = `${selNode.value.toLowerCase()}`;
+      if (!caseSensitivePseudoClassesSet.has(normalizeVal)) {
```
Comment (evilebottnawi) [inline src/selector.js]: Need lowercase `::before`/`::after` and other known pseudo element

Comment (anikethsaha) [reply src/selector.js]: They are case insensitive so if someone writes them in CAPITAL, then it's wrong CSS

Comment (evilebottnawi) [reply src/selector.js]: what is pseudo-elements is case sensitivity?

File: src/selector.js (hunk 1 above)
Comment (anikethsaha) [reply src/selector.js]: `walkTags` will traverse over all the html tags right. I guess this package can be added to `postcss-selector-parser` . What do you think?

File: src/selector.js (hunk 2 above)
Comment (anikethsaha) [reply src/selector.js]: Active, focus, focus-within, visited, hover doesn't work when they are in caps

Comment (anikethsaha) [reply src/selector.js]: Rest all are case insensitive

Comment (evilebottnawi) [reply src/selector.js]: Try:
```css
      div::BEFORE {
        content: "tets\6D";
      }
```

works fine

Comment (anikethsaha) [reply src/selector.js]: yeah but

```css
a:hover {
  color: red;
}
```

Wont work thats why I kept them in a list which are case sensitive

https://github.com/anikethsaha/postcss-lowercase-text/pull/3/files#diff-f67fc5ea4e003441c384bdf18e81ba52R3-R9

```css
div::before {
  content: "tets\6D";
}
```

Will get transformed to

```css
div::before {
  content: "tets\6D";
}
```

File: src/selector.js (hunk 1 above)
Comment (evilebottnawi) [reply src/selector.js]: Just add tests and check it, i am not manual :smile:

Comment (anikethsaha) [reply src/selector.js]: @evilebottnawi I added more tests https://github.com/anikethsaha/postcss-lowercase-text/pull/3/commits/a2065c5f65531d851309dc5ad4d516edab578689 and here https://github.com/anikethsaha/postcss-lowercase-text/pull/3/commits/652cd43349fad5df918c1ac0ee32707e9e9aadf8

**Pull request merged**